lib/spack/docs/packaging_guide.rst

```diff
@@ -553,6 +553,33 @@ version. This is useful for packages that have an easy to extrapolate URL, but
 keep changing their URL format every few releases. With this method, you only
 need to specify the ``url`` when the URL changes.
 
+"""""""""""""""""""""""
+Mirrors of the main URL
+"""""""""""""""""""""""
+
+Spack supports listing mirrors of the main URL in a package by defining
+the ``mirrors`` attribute. This attribute must return a list of URLs:
+
+.. code-block:: python
+
+  class Foo(Package):
+
+    url = "http://example.com/foo-1.0.tar.gz"
+    mirrors = [
+        'http://mirror.com/foo-1.0.tar.gz'
+    ]
```

Comment:
I'd rather make it so that `url` (or maybe `urls`) can be a list -- I'd rather have packages provide *one* attribute for this.  Specifically, for the above example, I think the syntax should look like this:

```python
  class Foo(Package):
      urls = [
          "http://example.com/foo-1.0.tar.gz",
          "http://mirror.com/foo-1.0.tar.gz",
      ]
```

Reply:
I agree with the renaming, `urls` would have been my first choice if I had no concerns on backward compatibility. What should we do? Use `url` and allow also lists or permit either `url` or `urls` to be present in a package?

Reply:
Now a package can have either the `url` or the `urls` attribute.